### PR TITLE
Add the rest of integration tests for labeling

### LINKED ISSUE #166: Prepare integration tests for labeling
**AC**
- Define what we want to test with integration, when it comes to the whole labeling feature (for example, checking if LabelDefinition has been created while adding a label with new key). Cover Label and LabelDefinition use cases.
- Write defined test cases
- Generate examples

### PR DESCRIPTION
**Pull request by dbadura**
**Description**
Add  the rest of required e2e labeling tests
Changes proposed in this pull request:
- search by labels (implemented, but turned off, because currently we don't filter on applications)
- list LDs
- one scenario required, could be not "default", when deleting last one, fail (FOR APPlication)
- Runtimes could have no labels (scenarios as well)

**Related issue(s)**
closes #166

### CODE REVIEW

File: tests/end-to-end/director/labeling_api_test.go [hunk 1]
```diff
@@ -563,3 +565,173 @@ func TestDeleteDefaultValueInScenariosLabelDefinition(t *testing.T) {
 	require.Error(t, err)
 	assert.EqualError(t, err, errMsg)
 }
+
+func TestSearchApplicationsByLabels(t *testing.T) {
+	// GIVEN
+	//TODO: enable this test after implementing filtering on applications
+	t.SkipNow()
```
Comment (pkosiec) [inline tests/end-to-end/director/labeling_api_test.go]: Please edit description of the task "DB for Applications domain". We need to remember to unskip it.

File: tests/end-to-end/director/labeling_api_test.go [hunk 2]
```diff
@@ -563,3 +565,173 @@ func TestDeleteDefaultValueInScenariosLabelDefinition(t *testing.T) {
 	require.Error(t, err)
 	assert.EqualError(t, err, errMsg)
 }
+
+func TestSearchApplicationsByLabels(t *testing.T) {
```
Comment (pkosiec) [inline tests/end-to-end/director/labeling_api_test.go]: What about searching Runtimes by labels? Could you please create such test?

File: tests/end-to-end/director/labeling_api_test.go [hunk 3]
```diff
@@ -563,3 +565,173 @@ func TestDeleteDefaultValueInScenariosLabelDefinition(t *testing.T) {
...
+	labelDefinitions, err := listLabelDefinitionsWithinTenant(t, ctx, tenantID)
+
+	//THEN
+	require.NoError(t, err)
+	require.Len(t, labelDefinitions, 2)
+	assert.Contains(t, labelDefinitions, firstLabelDefinition)
+	assert.Contains(t, labelDefinitions, secondLabelDefinition)
+}
+
+func TestDeletingLastScenarioForApplication_ShouldFail(t *testing.T) {
```
Comment (pkosiec) [inline tests/end-to-end/director/labeling_api_test.go]: Rename to `TestDeleteLastScenarioForApplication`

File: tests/end-to-end/director/labeling_api_test.go [hunk 4]
```diff
@@ -563,3 +565,173 @@ func TestDeleteDefaultValueInScenariosLabelDefinition(t *testing.T) {
...
+	appLabelRequest = fixSetApplicationLabelRequest(application.ID, scenariosLabel, []string{""})
+	appLabelRequest.Header["Tenant"] = []string{tenant}
+	err := tc.RunQuery(ctx, appLabelRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), `must be one of the following: "DEFAULT", "Christmas", "New Year"`)
+}
+
+func TestCreateRuntimeWithoutLabels(t *testing.T) {
```
Comment (pkosiec) [inline tests/end-to-end/director/labeling_api_test.go]: Shouldn't it be in Runtime test file? Or maybe even we test it already, as we check if runtime has correct properties (for example, no scenarios label if not provided)

Comment (dbadura) [reply tests/end-to-end/director/labeling_api_test.go]: All runtimes in `runtime_api_test.go` contains Labels.
Moved this to the `runtime_api_test.go`

File: tests/end-to-end/director/assertions.go [hunk 5]
```diff
@@ -193,6 +193,6 @@ func assertEventsAPI(t *testing.T, in []*graphql.EventAPIDefinitionInput, actual
 
 func assertRuntime(t *testing.T, in graphql.RuntimeInput, actual graphql.Runtime) {
 	assert.Equal(t, in.Name, actual.Name)
-	assert.Equal(t, *in.Description, *actual.Description)
+	assert.Equal(t, in.Description, actual.Description)
 	//assert.Equal(t, *in.Labels, actual.Labels) // TODO: Make it work when labels are in place
```
Comment (pkosiec) [inline tests/end-to-end/director/assertions.go]: shouldn't this be uncommented?

File: tests/end-to-end/director/labeling_api_test.go [hunk 6]
```diff
@@ -563,3 +565,230 @@ func TestDeleteDefaultValueInScenariosLabelDefinition(t *testing.T) {
...
+	labelDefinitions, err := listLabelDefinitionsWithinTenant(t, ctx, tenantID)
+
+	//THEN
+	require.NoError(t, err)
+	require.Len(t, labelDefinitions, 2)
+	assert.Contains(t, labelDefinitions, firstLabelDefinition)
+	assert.Contains(t, labelDefinitions, secondLabelDefinition)
+}
+
+func TestDeletingLastScenarioForApplication(t *testing.T) {
```
Comment (pkosiec) [inline tests/end-to-end/director/labeling_api_test.go]: `TestDelete`

**Pull request merged**